eva-server/src/test/java/uk/ac/ebi/eva/server/ws/ga4gh/GA4GHBeaconWSServerV2Test.java

```diff
@@ -0,0 +1,153 @@
+    }
+
+    @Test
+    public void testForExisting() throws Exception {
+        BeaconAlleleRequestBody request = new BeaconAlleleRequestBody();
+        request.setReferenceName("X");
+        request.setAssemblyId("GRCh37");
+        request.setReferenceBases("G");
+        request.setStart(new Long(100470026));
+        request.setEnd(new Long(100470026));
+        request.setAlternateBases("A");
+        request.setVariantType("SNV");
+        request.setDatasetIds(Arrays.asList("PRJEB7218"));
+        String url = String.format("/v2/beacon/query?referenceName=%s&referenceBases=%s&assemblyId=%s&alternateBases=" +
+                        "%s&start=%s&end=%s&variantType=%s&datasetIds=%s",
+                request.getReferenceName(),
+                request.getReferenceBases(),
+                request.getAssemblyId(),
+                request.getAlternateBases(),
+                request.getStart(),
+                request.getEnd(),
+                request.getVariantType(),
+                String.join(",", request.getDatasetIds()));
+
+        assertEquals(true, testBeaconHelper(url).getBody().getExists());
+
+        request.setStartMin(new Long(1));
+        request.setStartMax(new Long(1));
+        request.setEndMin(new Long(1));
+        request.setEndMax(new Long(1));
+        url = String.format("/v2/beacon/query?referenceName=%s&referenceBases=%s&assemblyId=%s&" +
+                        "alternateBases=%s&start=%s&end=%s&startMin=%s&endMin=%s&startMax=%s&endMax=%s&variantType=%s" +
+                        "&datasetIds=%s",
+                request.getReferenceName(),
+                request.getReferenceBases(),
+                request.getAssemblyId(),
+                request.getAlternateBases(),
+                request.getStart(),
+                request.getEnd(),
+                request.getStartMin(),
+                request.getEndMin(),
+                request.getStartMax(),
+                request.getEndMax(),
+                request.getVariantType(),
+                String.join(",", request.getDatasetIds()));
+        assertEquals(true, testBeaconHelper(url).getBody().getExists());
+    }
+
+    @Test
+    public void testForNonExisting() {
+        BeaconAlleleRequestBody request = new BeaconAlleleRequestBody();
+        request.setReferenceName("Y");
+        request.setAssemblyId("GRCh37");
+        request.setReferenceBases("G");
+        request.setAlternateBases("A");
+        String url = String.format("/v2/beacon/query?referenceName=%s&referenceBases=%s&assemblyId=%s&" +
+                        "alternateBases=%s",
+                request.getReferenceName(),
+                request.getReferenceBases(),
+                request.getAssemblyId(),
+                request.getAlternateBases());
+
+        assertEquals(false, testBeaconHelper(url).getBody().getExists());
+    }
+
+    /*
+        This test should fail because either alternateBases ot variantType is required
```

Comment:
Sorry to be so picky, but we try not to put redundant comments. I meant that *only if* it couldn't be expressed in code, then a comment is necessary. However, given that there's an assertion with a clear message, this comment is not needed :)

Reply:
That is perfectly fine, I will remove it :)

Reply:
fixed :)